src/simple_commander/game/game_controller.py

```diff
@@ -0,0 +1,190 @@
+""" This module describes behaviour of Game Controller. """
+
+import asyncio
+import json
+import logging
+from random import randint, shuffle
+
+from simple_commander.game.bullet import Bullet
+from simple_commander.game.invader import Invader
+from simple_commander.game.hero import Hero
+from simple_commander.utils.constants import STEP_INTERVAL, UNITS
+
+
+class GameController(object):
+    _instance = None
+    launched = False
+    websockets = {}
+
+    def __init__(self, height=None, width=None, invaders_count=None):
+        self.game_field = {'height': height, 'width': width}
+        self.invaders_count = invaders_count
+        self.units = {}
+        self.random_type = self.get_unit_type()
+        self.set_invaders(self.invaders_count)
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super(GameController, cls).__new__(cls)
+        return cls._instance
+
+    def action(self, data):
```

Comment:
What is action? This method name shares same named objects inside and makes a confuse to code readability.
Maybe we would rename it to `do_action()`.
Also `data` as an argument is accepted as the data to action call, but here we use the data as list of actions? a kind of mess